core/src/main/scala/org/apache/spark/deploy/ExternalShuffleService.scala

```diff
@@ -37,15 +39,17 @@ import org.apache.spark.util.Utils
  * Optionally requires SASL authentication in order to read. See [[SecurityManager]].
  */
 private[deploy]
-class ExternalShuffleService(sparkConf: SparkConf, securityManager: SecurityManager)
+class ExternalShuffleService(
+    sparkConf: SparkConf,
+    securityManager: SecurityManager,
+    transportConf: TransportConf,
```

Comment:
This interface is confusing: we pass in both a `SparkConf` and a `TransportConf`, even though one is created from another one.